### PR TITLE
clear decorations on window reload

### LINKED ISSUE #155446: Changing decorations to not show in the overview ruler sometimes doesn't hide some of them
Recording 2022-07-17 at 16 50 13

### PR DESCRIPTION
**Pull request by meganrogge**
fix #155446

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts [hunk 1]
```diff
@@ -108,6 +110,7 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 				}
 			}
 		}));
+		this._register(lifecycleService.onWillShutdown(() => this._disposeAllDecorations()));
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts:113]: Is `onWillShutdown` run when a window is closed or when all windows are closed?

**Review by TylerLeonhardt — DISMISSED** [review]

**Review by TylerLeonhardt — DISMISSED** [review]

File: src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts:113]: there's one `decorationAddon` created for each terminalInstance, so this should be fine 👍🏼 good thought though. works based on my testing

**Review by TylerLeonhardt — APPROVED** [review]

File: src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts (hunk 1 above)
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts:113]: I don't understand the fix here, was this problem only happening on reconnected terminal or something?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts:113]: yes

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts:113]: https://github.com/microsoft/vscode/issues/155446#issuecomment-1203155690

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts:113]: cool, thanks

**Pull request merged**